src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -827,17 +838,30 @@ class AccessibleBuffer extends DisposableStore {
 		}
 
 		if (!this._registered) {
-			this.add(this._terminal.raw.registerBufferElementProvider({ provideBufferElements: () => this._editorContainer }));
-			// When this is created, the element isn't yet attached so the dimensions are tiny
-			this._bufferEditor.layout({ width: this._accessibleBuffer.clientWidth, height: this._accessibleBuffer.clientHeight });
+			this._bufferEditor.layout({ width: xtermElement.clientWidth, height: xtermElement.clientHeight });
+			this._bufferEditor.onKeyDown((e) => {
+				if (e.keyCode === KeyCode.Escape || e.keyCode === KeyCode.Tab) {
+					this._accessibleBuffer.classList.remove('active');
+					this._terminal.raw.focus();
+					xtermElement.classList.remove('hide');
+				}
+			});
+			this.add(addDisposableListener(this._accessibleBuffer, 'keypress', (e) => {
+				if (e.key === 'Escape' || e.key === 'Tab') {
+					this._accessibleBuffer.classList.remove('active');
+					xtermElement.classList.remove('hide');
+					this._terminal.raw.focus();
+				}
+			}));
```

Comment:
Can these be regular vscode keybindings now?

Reply:
tried and seems we need them on the editor element itself for it to work